plugins/modules/zos_blockinfile.py

```diff
@@ -666,19 +666,27 @@ def main():
 
     marker = "{0}\\n{1}\\n{2}".format(marker_begin, marker_end, marker)
     block = transformBlock(block, ' ', indentation)
+    gdg = False
     # analysis the file type
+    if "/" not in src:
+        dataset = data_set.MVSDataSet(
+            name=src
+        )
+        src = dataset.name
+        gdg = dataset.name
+
+    if ("(" in src and ")" in src) and ("+" in src or "-" in src) and gdg is False:
```

Comment:
Couldn't you use `is_gds_positive_relative_name` or the other `is_gds_positive_relative_name` here to see if the user supplied a data set that doesn't exist yet?